File: UI/musicshop_admin/lib/models/guitar/guitar_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'guitar_insert_request.g.dart';

@JsonSerializable()
class GuitarInsertRequest {
  int? guitarTypeId;
  int? brandId;
  String? pickups;
  String? pickupConfiguration;
  String? description;
  String? model;
  int? frets;
  double? price;
  String? image;

  GuitarInsertRequest();

  factory GuitarInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$GuitarInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$GuitarInsertRequestToJson(this);
}

File: MusicShop.Model/Requests/GuitarInsertRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicShop.Model.Requests
{
    public class GuitarInsertRequest
    {
        public byte[]? ProductImage { get; set; }
        public int BrandId { get; set; }
        public string Model { get; set; }
        public decimal Price { get; set; }
        public string Description { get; set; }
        public int GuitarTypeId { get; set; }
        public string Pickups { get; set; }
        public string PickupConfiguration { get; set; }
        public int Frets { get; set; }
    }
}
